src/pkg/services/m365/api/drive_pager_test.go

```diff
@@ -199,7 +199,8 @@ func (suite *DrivePagerIntgSuite) TestEnumerateDriveItems() {
 				Select: api.DefaultDriveItemProps(),
 			})
 
-	for page, reset, done := pager.NextPage(); !done; {
+	page, reset, done := pager.NextPage()
```

Comment:
I'm curious as to why these variables need to exist outside the loop? I don't see any references to them in the code after the loop

Reply:
It's equivalent to:
```go
	for page, reset, done := pager.NextPage();
	!done;
	page, reset, done = pager.NextPage() {
```
or 
```go
page, reset, done := pager.NextPage()
for !done {
	//...
	page, reset, done = pager.NextPage()
}
```

The format I originally used (`for a, b, c := foo(); !c; {`) was a bug; it instantiates the values, but doesn't update them.  This just felt like the quickest fix.  chonky, but workable.

Reply:
ah, I see. You can do the update in the loop as well. It just turns into
```go
for page, reset, done := pager.NextPage(); !done; page, reset, done = pager.NextPage() {
```

Reply:
Dear santa, all I want for christmas is `for foo, bar, keepRanging := range baz() {`.

Reply:
also, wow, the linter does _not_ like the multiline style in my first example.